### PR TITLE
Complete the function get_hw_loop_zone_list_w_area

### PR DESCRIPTION
**Pull request by weilixu**
Function completed for Rule 21-5 and 21-10

### CODE REVIEW

File: rct229/ruleset_functions/get_hw_loop_zone_list_w_area.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            ]
+        )
+        for terminal in find_all("$.terminals[*]", zone):
+            hhw_loop_id = None
+            if terminal.get("heating_source") == HEATING_SOURCE.HOT_WATER:
+                # if terminal has reheat.
+                # If a terminal heating source is hot water, then it must have a heating_from_loop
+                # Otherwise, raise exception
+                hhw_loop_id = getattr_(terminal, "terminal", "heating_from_loop")
+            if not hhw_loop_id and terminal.get(
```
Comment (charliepnnl) [inline rct229/ruleset_functions/get_hw_loop_zone_list_w_area.py:48]: If the condition on line 43 is True, then hhw_loop_id will be set, so line 48 could be
```
elif teminal.get("served_by_heating_ventilation_air_conditioning_system"):
```

File: rct229/ruleset_functions/get_hw_loop_zone_list_w_area.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                if is_hvac_sys_preheating_type_fluid_loop(rmi_b, hvac_id):
+                    # check the case the terminal is connected with an HVAC, whose preheating system is supplied
+                    # by a hot water loop. The is_hvac_sys_preheating_type_fluid_loop returns true only
+                    # 1. preheat_system exist and,
+                    # 2. preheat_system.hot_water_loop exist and,
+                    # 3. preheat_system.heating_system_type is HEATING_SYSTEM.FLUID_LOOP
+                    hhw_loop_id = find_exactly_one_hvac_system(rmi_b, hvac_id)[
+                        "preheat_system"
+                    ]["hot_water_loop"]
+                if not hhw_loop_id and is_hvac_sys_heating_type_fluid_loop(
```
Comment (charliepnnl) [inline rct229/ruleset_functions/get_hw_loop_zone_list_w_area.py:63]: Similar to above, if the condition on line 54 is True, then `hhw_loop_id` will be set, so line 63 could be
```
if is_hvac_sys_heating_type_fluid_loop(rmi_b, hvac_id):
```

**Review by charliepnnl — APPROVED** [review]
There are some comments for your consideration.

@weilixu Are we renaming functions that return a dictionary by appending `_dict` or not?  I've done this, but we haven't done it consistently.

**Review by yunjoonjung-PNNL — APPROVED** [review]
Looks great to me!

**Comment by weilixu** [comment]
> There are some comments for your consideration.
> 
> @weilixu Are we renaming functions that return a dictionary by appending `_dict` or not? I've done this, but we haven't done it consistently.

I will rename it. Thanks

File: rct229/ruleset_functions/get_hw_loop_zone_list_w_area.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        )
+        for terminal in find_all("$.terminals[*]", zone):
+            hhw_loop_id = None
+            if terminal.get("heating_source") == HEATING_SOURCE.HOT_WATER:
+                # if terminal has reheat.
+                # If a terminal heating source is hot water, then it must have a heating_from_loop
+                # Otherwise, raise exception
+                hhw_loop_id = getattr_(terminal, "terminal", "heating_from_loop")
+            if not hhw_loop_id and terminal.get(
+                "served_by_heating_ventilation_air_conditioning_system"
```
[the remark this answers is not in the document]
Comment (charliepnnl) [reply rct229/ruleset_functions/get_hw_loop_zone_list_w_area.py]: Ok

**Pull request merged**